### PR TITLE
[prometheus] make path.rootfs conditional

### LINKED ISSUE #467: [kube-prometheus-stack] node-exporter crashes at startup
**Describe the bug**
Node exporter pod underlying container fails to startup. This is the error reported by the deamon:

``` console
Error: failed to start container "node-exporter": Error response from daemon: path / is mounted on / but it is not a shared or slave mount
```

This error is related to #325 and it appears to be caused by this [pull request](https://github.com/prometheus-community/helm-charts/pull/80#issuecomment-728716134). I've create this issue as suggested by @monotek in https://github.com/prometheus-community/helm-charts/issues/80#issuecomment-728922569 (it appears that the original reporter of this issue has not created a new issue).

Moreover, I've tried to change kube-prometheus-stack chart manually to revert the commit above and I am able to deploy the chart without any error.

I am using `docker-desktop` (v2.5.0.1) on Windows 10 1909 and I have enabled Kubernetes cluster from it.

I'm not sure if this error is caused by `docker-desktop`, in any case, if it does not happen in other type of clusters, I would suggest to make the **mount rootfs** *mountPropagation* configurable via the chart values (and maybe enable it by default if you think it should not be optional) to prevent this error.

Another (better) solution would be to mount / as shared, but I've not found a way to do it.

**Version of Helm and Kubernetes**:

Helm Version:

```console
version.BuildInfo{Version:"v3.4.1", GitCommit:"c4e74854886b2efe3321e185578e6db9be0a6e29", GitTreeState:"clean", GoVersion:"go1.14.11"}
```

Kubernetes Version:

```console
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.1", GitCommit:"7879fc12a63337efff607952a323df90cdc7a335", GitTreeState:"clean", BuildDate:"2020-04-08T17:38:50Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"windows/amd64"}
Server Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.3", GitCommit:"1e11e4a2108024935ecfcb2912226cedeafd99df", GitTreeState:"clean", BuildDate:"2020-10-14T12:41:49Z", GoVersion:"go1.15.2", Compiler:"gc", Platform:"linux/amd64"}
```

**Which chart**: kube-prometheus-stack

**Which version of the chart**: 12.7.0

**What happened**:  node-exporter pod fails to startup

**What you expected to happen**: node-exporter should not fail

**How to reproduce it**:

No additional change in values.yaml file to reproduce.

**The helm command that you execute and failing/misfunctioning**:

```console
helm install prom-stack prometheus-community/kube-prometheus-stack --version v12.7.0 -n monitoring
```

### PR DESCRIPTION
**Pull request by radhus**
#### What this PR does / why we need it:

Makes it configurable if node-exporter daemonset should mount `/` from host or not. Default to `true` to keep existing default behavior.

As seen in issue #467, it is not always possible to mount / from host in all environments.

#### Which issue this PR fixes

  - fixes #467 (but perhaps needs similar change to the node-exporter chart as well?)

#### Special notes for your reviewer:

> For fast feedback, please @-mention maintainers that are listed in the Chart.yaml file.
@gianrubio , @zanhsieh , @Xtigyro , @monotek , @naseemkullah 

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/prometheus/Chart.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v2
 name: prometheus
-version: 13.4.0
+version: 13.4.1
```
Comment (monotek) [inline charts/prometheus/Chart.yaml:3]: New feature should raise minor version.

**Review by monotek — CHANGES_REQUESTED** [review]

File: charts/prometheus/Chart.yaml (hunk 1 above)
Comment (radhus) [reply charts/prometheus/Chart.yaml]: Fixed, bumped to `13.5.0`

**Review by monotek — CHANGES_REQUESTED** [review]
Version needs to be 13.6.0

**Comment by radhus** [comment]
@monotek bumped it, I'd usually like to rebase the PR but the template says

> When updates to your PR are requested, please add new commits and do not squash the history.

Let me know if I shall proceed and do that or if it can be merged anyway.

**Review by monotek — DISMISSED** [review]

**Review by zanhsieh — APPROVED** [review]

**Comment by radhus** [comment]
Anything else needed to get this merged? 🙂

**Review by monotek — APPROVED** [review]

**Pull request merged**